src/pages/_app.jsx

```diff
@@ -1,95 +1,80 @@
+import { wrapper } from '../redux/store';
 import '../../assets/self-styles.less';
 import '../../assets/nprogress.css';
+import configure from '../utils/amplify-config';
+import CustomError from '../utils/customError';
 
-import { isBrowser, getCurrentEnvironment } from '../utils/environment';
-import setupAmplify from '../utils/setupAmplify';
+// import { isBrowser } from '../utils/environment';
+// TODO: this needs to be refactored after auth works properly
+//
+// import { isBrowser, getCurrentEnvironment } from '../utils/environment';
+// import setupAmplify from '../utils/setupAmplify';
 
 Router.events.on('routeChangeStart', () => NProgress.start());
 Router.events.on('routeChangeComplete', () => NProgress.done());
 Router.events.on('routeChangeError', () => NProgress.done());
 
-const WrappedApp = ({ Component, pageProps, req }) => {
+const WrappedApp = ({ Component, pageProps }) => {
+  const { httpError } = pageProps;
   const router = useRouter();
-  const [experimentId, setExperimentId] = useState(undefined);
 
-  console.log(req, pageProps);
+  const { auth } = useSelector((state) => state.networkResources);
+
+  const { experimentId } = router.query;
+  const experimentData = useSelector((state) => (experimentId ? state.experimentSettings.info : {}));
 
-  // Only hydrate pages when experiment ID is loaded.
   useEffect(() => {
-    if (!router.route.includes('experimentId')) {
-      setExperimentId(false);
-    }
+    const { userPoolId, identityPoolId, userPoolClientDetails } = auth;
 
-    if (router.route.includes('experimentId')) {
-      setExperimentId(router.query.experimentId);
+    if (!auth.userPoolId || !auth.identityPoolId) {
+      return;
     }
 
-    setupAmplify(getCurrentEnvironment());
-  }, [router.query.experimentId]);
-
-  const { data: experimentData, error: experimentError } = useSWR(
-    () => (experimentId ? `/v1/experiments/${experimentId}` : null),
-    getFromApiExpectOK,
-  );
+    Amplify.configure({
+      ...configure(userPoolId, identityPoolId, userPoolClientDetails),
+      ssr: true,
+    });
+  }, [auth]);
 
   const mainContent = () => {
-    // If the page is statically rendered (on server), show a loading context.
-    if (!isBrowser) {
-      return (
-        <PreloadContent />
-      );
-    }
-
     // If this is a not found error, show it without the navigation bar.
     if (Component === NotFoundPage) {
       return <Component {...pageProps} />;
     }
 
-    // If the experiment ID does not exist and is not `false` (i.e. not necessary)
-    // wait until the experiment ID is loaded before loading the page.
-    if (!experimentId && experimentId !== false) {
-      return (<PreloadContent />);
-    }
-
-    // If we found the experiment ID, but we haven't yet queried the API for data
-    // about the experiment, wait until that is done.
-    if (experimentId && !experimentData && !experimentError) {
-      return (<PreloadContent />);
-    }
-
```

Comment:
Experiment data loading is now server-side, no need for these as page won't load until the data is available.